node/connection_base.js

```diff
@@ -104,6 +131,39 @@ TChannelConnectionBase.prototype.resetAll = function resetAll(err) {
         err = new Error('unknown connection reset'); // TODO typed error
     }
 
+    if (!self.remoteName) {
+        if (self.direction === 'out') {
+            self.channel.connectionsConnectErrorsStat.increment(1, {
+                'host-port': self.channel.hostPort || '0.0.0.0:0',
+                'peer-host-port': self.remoteAddr
+            });
+        } else {
+            self.channel.connectionsAcceptedErrorsStat.increment(1, {
+                'host-port': self.channel.hostPort
+            });
+        }
+    } else {
+        if (err.type !== 'tchannel.socket-local-closed' &&
+            err.type !== 'tchannel.socket-closed') {
```

Comment:
socket-closed is also an error. The only one's that are not errors are local closes.